apps/storage/src/app.ts

```diff
@@ -0,0 +1,565 @@
+import { getCookie } from 'hono/cookie';
+import { serve } from '@hono/node-server';
+import {
+  convoAttachments,
+  orgMemberProfiles,
+  orgMembers,
+  orgs,
+  pendingAttachments,
+  teams
+} from '@u22n/database/schema';
+import {
+  typeIdGenerator,
+  typeIdValidator,
+  type TypeId
+} from '@u22n/utils/typeId';
+import { and, eq } from '@u22n/database/orm';
+import { db } from '@u22n/database';
+import { zValidator } from '@hono/zod-validator';
+import { storage } from './storage';
+import { env } from './env';
+import { z } from 'zod';
+import {
+  DeleteObjectsCommand,
+  GetObjectCommand,
+  PutObjectCommand
+} from '@aws-sdk/client-s3';
+import { getSignedUrl } from '@aws-sdk/s3-request-presigner';
+import { s3Client } from './s3';
+import { File } from 'buffer';
+import sharp from 'sharp';
+
+type Variables = {
+  account: {
+    id: number;
+    session: any;
+  } | null;
+};
+
+const app = new Hono<{ Variables: Variables }>();
+
+// CORS middleware
+app.use(
+  '*',
+  cors({
+    origin: env.WEBAPP_URL,
+    credentials: true
+  })
+);
+
+// Auth middleware
+app.use('*', async (c, next) => {
+  const sessionCookie = getCookie(c, 'unsession');
+  if (!sessionCookie) {
+    c.set('account', null);
+    await next();
+  } else {
+    const sessionObject = await storage.getItem(sessionCookie);
+    if (!sessionObject) {
+      c.set('account', null);
+      await next();
+    } else {
+      c.set('account', {
+        // @ts-expect-error, not typed properly yet
+        id: Number(sessionObject.attributes.account.id),
+        session: sessionObject
+      });
+      await next();
+    }
+  }
+});
+
+// Middleware to check if user is signed in
+const checkSignedIn = async (
+  c: Context<{ Variables: Variables }>,
+  next: Next
+) => {
+  if (!c.get('account'))
+    return c.json({ error: 'Unauthorized' }, { status: 401 });
+  await next();
+};
+
+// Middleware to check if service is authorized (aka intra-service communication)
+const checkAuthorizedService = async (
+  c: Context<{ Variables: Variables }>,
+  next: Next
+) => {
+  const authToken = c.req.header('Authorization');
+  if (!authToken || authToken !== env.STORAGE_KEY)
+    return c.json({ error: 'Unauthorized' }, { status: 401 });
+  await next();
+};
+
+// 404 handler
+app.notFound((c) => c.json({ message: 'Not Found' }, 404));
+
+// Proxy /avatar to `${process.env.STORAGE_S3_ENDPOINT}/${process.env.STORAGE_S3_BUCKET_AVATARS}/${proxy}`
+app.get('/avatar/:proxy{.+}', async (c) => {
+  const proxy = c.req.param('proxy');
+  const res = await fetch(
+    `${env.STORAGE_S3_ENDPOINT}/${env.STORAGE_S3_BUCKET_AVATARS}/${proxy}`
+  );
+  if (res.status === 404) {
+    return c.json({ error: 'Not Found' }, { status: 404 });
+  }
+  // Copy response as fetch headers are immutable
+  return new Response(res.body, res);
+});
+
+app
+  // Health check endpoint
+  .get('/', (c) => c.json({ status: "I'm Alive 🏝️" }))
+  // Attachment download endpoint
+  .get(
```

Comment:
why is this chained?
I would rather it was app.get per route to make it easier to skim through